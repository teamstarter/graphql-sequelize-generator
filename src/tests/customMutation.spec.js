const request = require('supertest')
const { deleteTables } = require('./testDatabase.js')
const { createServer, closeServer, resetDb } = require('./setupTestServer')

/**
 * Starting the tests
 */
describe('Test the custom mutations', () => {
  let server = null
  let trace = []
  const globalPreCallback = (type) => {
    trace.push(type)
  }

  beforeAll(async () => {
    server = await createServer({}, globalPreCallback)
  })

  afterAll(() => closeServer(server))

  beforeEach(async () => {
    trace = []
    await resetDb()
  })

  afterEach(async () => {
    await deleteTables()
  })

  it('Check that you can call a custom Mutation', async () => {
    const responseUserCreate = await request(server)
      .post('/graphql')
      .set('userid', 1)
      .send({
        query: `mutation logThat($message: String) {
              logThat(message: $message) {
                message
              }
            }`,
        variables: {
          message: 'Hello',
        },
        operationName: 'logThat',
      })
    expect(responseUserCreate.body.errors).toBeUndefined()
    expect(responseUserCreate.body.data.logThat).not.toBeUndefined()
    expect(responseUserCreate.body.data.logThat).toMatchSnapshot(
      'The sent message'
    )
    expect(trace).toMatchSnapshot()
  })

  it('Check that you can call an additional mutation on a model', async () => {
    const response = await request(server)
      .post('/graphql')
      .set('userid', 1)
      .send({
        query: `mutation anonymizeLog($logId: String) {
              anonymizeLog(logId: $logId) {
                id
                message
              }
            }`,
        variables: {
          logId: '123',
        },
        operationName: 'anonymizeLog',
      })
    expect(response.body.errors).toBeUndefined()
    expect(response.body.data.anonymizeLog).not.toBeUndefined()
    expect(response.body.data.anonymizeLog).toMatchSnapshot(
      'The anonymized log'
    )
    expect(trace).toMatchSnapshot()
  })
})
