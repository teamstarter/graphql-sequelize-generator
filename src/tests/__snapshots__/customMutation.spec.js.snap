// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the custom mutations Check that you can call a custom Mutation 1`] = `
Array [
  "logThatCustomResolver",
]
`;

exports[`Test the custom mutations Check that you can call a custom Mutation: The sent message 1`] = `
Object {
  "message": "Hello",
}
`;

exports[`Test the custom mutations Check that you can call an additional mutation on a model 1`] = `Array []`;

exports[`Test the custom mutations Check that you can call an additional mutation on a model: The anonymized log 1`] = `
Object {
  "id": 1,
  "message": "Anonymized",
}
`;
